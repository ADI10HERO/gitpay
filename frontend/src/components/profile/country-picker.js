import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

class CountryPicker extends Component {
  static propTypes = {
    //classes: PropTypes.object.isRequired,
    open: PropTypes.bool,
    onClose: PropTypes.func
  }

  render() {
    return (
      <div>
        <Dialog
          open={this.props.open}
          onClose={this.props.onClose}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <DialogTitle id="alert-dialog-title">{"Choose your country"}</DialogTitle>
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
            Please choose the country that you have your bank account to receive your bounties when conclude any task
            </DialogContentText>
            <DialogContentText id="alert-dialog-flags">
              
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={this.props.onClose} color="primary">
              Cancel
            </Button>
            <Button onClick={this.props.onClose} color="primary" autoFocus>
              Confirm
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }

}

export default CountryPicker