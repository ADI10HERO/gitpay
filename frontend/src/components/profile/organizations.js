import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Chip from '@material-ui/core/Chip'
import TagFacesIcon from '@material-ui/icons/TagFaces'

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    padding: theme.spacing.unit / 2,
  },
  chip: {
    margin: theme.spacing.unit / 2,
  },
})

class Organizations extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    data: PropTypes.object
  }

  handleDelete = data => () => {
    if (data.label === 'React') {
      console.log('Why would you want to delete React?! :)') // eslint-disable-line no-console
      return
    }

    this.setState(state => {
      const chipData = [...state.chipData]
      const chipToDelete = chipData.indexOf(data)
      chipData.splice(chipToDelete, 1)
      return { chipData }
    })
  };

  render () {
    const { classes, data } = this.props

    return (
      <div className={ classes.root }>
        { data.map(data => {
          let icon = null

          if (data.label === 'React') {
            icon = <TagFacesIcon />
          }

          return (
            <Chip
              key={ data.key }
              icon={ icon }
              label={ data.label }
              onDelete={ this.handleDelete(data) }
              className={ classes.chip }
            />
          )
        }) }
      </div>
    )
  }
}

export default withStyles(styles)(Organizations)
