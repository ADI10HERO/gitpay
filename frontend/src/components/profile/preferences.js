import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl'

import Paper from 'material-ui/Paper'
import Typography from 'material-ui/Typography'

const languages = {
  'br': 'PortuguÃªs',
  'en': 'English'
}

class Preferences extends Component {

  render () {
    return (
      <Paper elevation={ 0 }>
        <Typography variant='headline' component='h3'>
          <FormattedMessage id='preferences.headline' />
        </Typography>
        <Typography component='p' style={ { marginBottom: 40 } }>
          <FormattedMessage id='preferences.subtitle' />
        </Typography>
        <Typography component='p' style={ { marginBottom: 10 } }>
          <strong><FormattedMessage id='preferences.language.name' /></strong> { languages[this.props.preferences.language] }
        </Typography>
        <Typography component='p' style={ { marginBottom: 40 } }>
          <strong><FormattedMessage id='preferences.language.currency' /></strong> <FormattedMessage id='preferences.language.currency.name' />
        </Typography>
      </Paper>
    )
  }
}

export const PreferencesPure = Preferences
export default withRouter(Preferences)
