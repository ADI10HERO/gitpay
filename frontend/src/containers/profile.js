import { connect } from 'react-redux'
import Profile from '../components/profile/profile'

const mapStateToProps = (state, ownProps) => {
  return {
    logged: state.loggedIn.logged,
    user: state.loggedIn.user,
    preferences: state.preferences,
    completed: state.loggedIn.completed
  }
}

export default connect(mapStateToProps)(Profile)
