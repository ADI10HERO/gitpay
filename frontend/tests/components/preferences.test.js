import React from 'react'
import { PreferencesPure } from '../../src/components/profile/preferences'
import { configure } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'

configure({ adapter: new Adapter() })

describe('components', () => {
  describe('Preferences component', () => {
    it('should start a new preferences with a defined language', () => {
      const preferencesComponent = () => (<PreferencesPure preferences={ { language: 'br' } } />)
      const component = renderer.create(preferencesComponent()).toJSON()
      console.log(component)
      expect(component).toMatchSnapshot()
    })
  })
})
