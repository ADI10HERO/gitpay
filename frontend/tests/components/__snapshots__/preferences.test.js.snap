// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Preferences component should start a new preferences with a defined language 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-shadow0-3 MuiPaper-rounded-2"
>
  <h3
    className="MuiTypography-root-28 MuiTypography-headline-33"
  />
  <p
    className="MuiTypography-root-28 MuiTypography-body1-37"
    style={
      Object {
        "marginBottom": 40,
      }
    }
  />
  <p
    className="MuiTypography-root-28 MuiTypography-body1-37"
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <strong />
     
    PortuguÃªs
  </p>
  <p
    className="MuiTypography-root-28 MuiTypography-body1-37"
    style={
      Object {
        "marginBottom": 40,
      }
    }
  >
    <strong />
  </p>
</div>
`;
